helpers/PermissionManager.js

```diff
@@ -247,6 +259,7 @@ module.exports.getUserPermissions = async function(username, provider, webstrate
 		snapshot);
 
 	setCachedPermissions(username, provider, permissions, snapshot.id);
+	if (await userMustBeLoggedInToWriteAndUserIsAnonymous(username, provider)) permissions = permissions.replace("w", "");
```

Comment:
Same as above.

Reply:
Also, consider that if the `permissions` string was accidentally (or maliciously) set to `rww`, `permissions.replace("w", "");` would only remove the first occurrence, i.e. `permissions` would afterwards be `rw`, and the webstrate thus still annonymously writeable, regardless of the `loggedInToWrite` to write flag.

This potential problem could be avoided by doing `permissions.replace(/w/g, '')`.

Reply:
I changed it to `permissions.replace(/w/g, '')`